src/components/coverage-date-list/coverage-date-list.js

```diff
@@ -3,9 +3,38 @@ import PropTypes from 'prop-types';
 import moment from 'moment';
 import { FormattedDate, FormattedMessage } from 'react-intl';
 
+/* eslint-disable consistent-return */
+const containsNonEmptyObjectsWithStringValues = (propValue, key, componentName, location, propFullName) => {
+  const error = new Error(
+    'Invalid prop `' + propFullName + '` supplied to' +
+    ' `' + componentName + '`. Validation failed.'
+  );
+
+  const BEGIN_COVERAGE = 'beginCoverage';
+  const END_COVERAGE = 'endCoverage';
+
+  const curObject = propValue[key];
+  const curObjectKeys = Object.keys(curObject);
+  const containsBeginCoverageProp = curObjectKeys.includes(BEGIN_COVERAGE);
+  const containsEndCoverageProp = curObjectKeys.includes(END_COVERAGE);
+
+  if (!containsBeginCoverageProp && !containsEndCoverageProp) {
+    return error;
+  }
+
+  if (containsBeginCoverageProp && typeof curObject.beginCoverage !== 'string') {
+    return error;
+  }
+
+  if (containsEndCoverageProp && typeof curObject.endCoverage !== 'string') {
+    return error;
+  }
```

Comment:
possibly it will be better to refactor this to (u can use other variable name):
```suggestion
const isError =  (!containsBeginCoverageProp && !containsEndCoverageProp)
        || (containsBeginCoverageProp && typeof curObject.beginCoverage !== 'string') 
        || (containsEndCoverageProp && typeof curObject.endCoverage !== 'string') ;
        
  if (isError) {
     return error;
   }
```